### PR TITLE
[wip] LOD generation

### PR DESCRIPTION
**Pull request by Shubhamai**

### CODE REVIEW

File: threedgrut/strategy/mcmc.py [hunk 1]
```diff
@@ -92,13 +92,36 @@ def post_optimizer_step(self, step: int, scene_extent: float, train_dataset, bat
...
+        
+        dead_idxs = dead_idxs[dead_idxs >= self.conf.lods.prev_n_gaussians]
+        alive_idxs = alive_idxs[alive_idxs >= self.conf.lods.prev_n_gaussians]
+
         n_dead_gaussians = len(dead_idxs)
 
         if n_dead_gaussians:
             sampled_idxs, new_densities, new_scales = self.sample_new_gaussians(n_dead_gaussians, alive_idxs)
+
+            # Check for invalid densities: squeeze to (N), then check for nan/inf
```
Comment (Shubhamai) [inline threedgrut/strategy/mcmc.py:108]: This is a hack as because were some nan being generated in `new_densities` or `new_scales`  causing loss to be `nan`

Comment (Shubhamai) [reply threedgrut/strategy/mcmc.py:108]: Not sure why `self.model.get_scale()` sometimes return `nan` in `sample_new_gaussians`
